EventProc/include/EventProc/OverlayProducer.h

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file OverlayProducer.h
+ * @brief Class to overlay in-time pile-up events from an overlay file 
+ * @author Lene Kristian Bryngemark, Stanford University 
+ */
+
+#ifndef EVENTPROC_OVERLAYPRODUCER_H
+#define EVENTPROC_OVERLAYPRODUCER_H
+
+//ROOT
+#include "TFile.h"
+#include "TRandom2.h"
+
+// STL
+#include <string>
+#include <vector>
+#include <map>
+
+//LDMX Framework
+#include "Event/EventDef.h" // Includes all possible event bus objects
+#include "Framework/EventFile.h"
+#include "Framework/EventProcessor.h" // Needed to declare processor
+#include "Framework/Parameters.h" // Needed to import parameters from configuration file
+
+
+namespace ldmx {
+    
+    /**
+     * @class OverlayProducer
+     * @brief 
+     */
+  class OverlayProducer : public ldmx::Producer {
+  public:
+
+	OverlayProducer(const std::string& name, ldmx::Process& process) : ldmx::Producer(name, process), overlayEvent_{"overlay"}{  }
+
+	//	~OverlayProducer();
+	//destructor causes linking problems 
+	
+	virtual void configure(Parameters& parameters) final override;
+
+	virtual void produce(Event& event) final override;
+
+	virtual void onProcessStart(); 
+
+  private:
+
+
+	/**
+	 * Pileup overlay events input file name
+	 */
+	std::string overlayFileName_;
+
+	/** 
+	 * Pileup overlay events input file
+	 */
+	std::unique_ptr<EventFile> overlayFile_;      
+	    
+	/**
+	 * The overlay ldmx event bus
+	 */
+	Event overlayEvent_;
+
+	/**
+	 * List of simhit collection(s) to loop over and add hits from, combining sim and pileup
+	 */
+	std::vector <std::string> collections_;     
+
+	/**
+	 * Pileup overlay events input pass name
+	 */
+	std::string overlayPassName_;   
+
+	/**
+	 * To use for finding the sim event bus passengers, mostly a disambiguation
+	 */
+	std::string simPassName_;
+	
+	/**
+	 * Let the total number of events be poisson distributed, or fix at the chosen value, poissonMu_
+	 */
+	int doPoisson_{0};
```

Comment:
We are now able to pass booleans as parameters. I would suggest using that capability unless this variable is used for something else that requires it to be an int.

Reply:
done